### PR TITLE
Developer

### PR DESCRIPTION
**Pull request by SonOfViktor**

### CODE REVIEW

File: persistence/src/main/java/com/epam/esm/config/DaoConfig.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+import javax.sql.DataSource;
+
+@Configuration
+@ComponentScan(basePackages = "com.epam.esm")
+//@Profile("prod")
```
Comment (KseniyaYudzichavaTR) [inline persistence/src/main/java/com/epam/esm/config/DaoConfig.java:15]: no comments

File: service/src/main/java/com/epam/esm/builder/SelectSqlBuilder.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+        String and = (!name.isEmpty() || !description.isEmpty() && !tagName.isEmpty()) ? AND : EMPTY_LINE;
+
+        if (!name.isEmpty() || !description.isEmpty() || !tagName.isEmpty()) {
+            String wherePart = String.format(pattern, tagName, and, name, or, description);
+            sql.append(wherePart);
+        }
+    }
+
+    private void appendOrderPart(StringBuilder sql, SelectParams params) {
```
Comment (KseniyaYudzichavaTR) [inline service/src/main/java/com/epam/esm/builder/SelectSqlBuilder.java:95]: return value from method - do not change incoming parameter

File: service/src/main/java/com/epam/esm/validator/SelectParameterValidator.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package com.epam.esm.validator;
+
+import com.epam.esm.entity.SelectParams;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SelectParameterValidator {
```
Comment (KseniyaYudzichavaTR) [inline service/src/main/java/com/epam/esm/validator/SelectParameterValidator.java:7]: rename class please

Comment (SonOfViktor) [reply service/src/main/java/com/epam/esm/validator/SelectParameterValidator.java:7]: Class was deleted after adding Apache commons lang3

File: persistence/src/main/java/com/epam/esm/config/DaoConfig.java (hunk 1 above)
Comment (SonOfViktor) [reply persistence/src/main/java/com/epam/esm/config/DaoConfig.java:15]: Done.

**Pull request merged**